### ISSUE TITLE
Character ranges in TweetTokenizer regular expressions

### ISSUE THREAD

**Issue by erickrf** [nltk/nltk#1183]
As reported in the nltk-dev mailing list, I found that the TweetTokenizer splits words at accented characters. Checking the code, I found at line 159:

```
(?:[a-z][a-z'\-_]+[a-z])       # Words with apostrophes or dashes.
```

A simple solution would be replacing the character ranges by `\w`:

```
(?:\w[\w'\-_]+\w)       # Words with apostrophes or dashes.
```

**Comment by stevenbird** [comment]
@fievelk – would you mind taking a look at this please?

**Comment by fievelk** [comment]
@stevenbird Sure. 
@erickrf Thank you for pointing this out.

**Comment by fievelk** [comment]
@erickrf could you please provide some examples please? I am not able to reproduce your issue using common accented characters.

**Comment by erickrf** [comment]
Sure:

```
tokenizer = nltk.tokenize.TweetTokenizer()
tokens = tokenizer.tokenize(u'testing tokenizer with resumé, españa, münchen, россия')
for token in tokens: print token,
```

Output:

```
testing tokenizer with resum é , espa ña , münchen , россия
```

The regexp groups the first three letters, accented or not. The splits occurs when the accents are after the first three.

**Comment by fievelk** [comment]
**Edit:** I am now able to reproduce your results:

```
>>> import nltk
>>> tokenizer = nltk.tokenize.TweetTokenizer()
>>> tokenizer.tokenize(u'testing tokenizer with resumé, españa, münchen, россия')
['testing', 'tokenizer', 'with', 'resum', 'é', ',', 'espa', 'ña', ',', 'münchen', ',', 'россия']
```

I will make a pull request soon.

**Comment by ewan-klein** [comment]
Resolved in 2f7acb81a496044077af934817e02616c76badce
